tools/c7n_azure/c7n_azure/resources/entraid_group.py

```diff
@@ -0,0 +1,518 @@
+
+            for member in members:
+                # Only analyze users (not other groups or service principals)
+                if member.get('@odata.type') == '#microsoft.graph.user':
+                    user_type = member.get('userType', 'Member')
+                    user_principal_name = member.get('userPrincipalName', '')
+
+                    # Check if user is a guest
+                    if user_type.lower() == 'guest':
+                        has_guest_members = True
+
+                    # Check if user is external (from different domain)
+                    # External users typically have #EXT# in their UPN or are guests
+                    if '#EXT#' in user_principal_name or user_type.lower() == 'guest':
+                        has_external_members = True
+
+            return {
+                'has_external_members': has_external_members,
+                'has_guest_members': has_guest_members,
+                'total_members': len([
+                    m for m in members
+                    if m.get('@odata.type') == '#microsoft.graph.user'
+                ])
+            }
+        except requests.exceptions.RequestException as e:
+            if "403" in str(e) or "Insufficient privileges" in str(e):
+                log.warning(
+                    f"Insufficient privileges to analyze member types for group {group_id}. "
+                    "Required permissions: GroupMember.Read.All, User.Read.All"
+                )
+
+                return None  # Unknown member types
+            else:
+                log.error(f"Error analyzing member types for group {group_id}: {e}")
+                return None
+
+
+@EntraIDGroup.filter_registry.register('member-count')
+class MemberCountFilter(Filter):
+    """Filter groups based on member count.
+
+    Required permission: GroupMember.Read.All
+
+    :example:
+
+    Find groups with too many members:
+
+    .. code-block:: yaml
+
+        policies:
+          - name: large-groups
+            resource: azure.entraid-group
+            filters:
+              - type: member-count
+                count: 100
+                op: greater-than
+    """
+
+    schema = type_schema(
+        'member-count',
+        count={'type': 'number'},
+        op={'type': 'string', 'enum': ['greater-than', 'less-than', 'equal']}
+    )
+
+    def process(self, resources, event=None):  # pylint: disable=unused-argument
+        count_threshold = self.data.get('count', 0)
+        op = self.data.get('op', 'greater-than')
+
+        filtered = []
+        for resource in resources:
+            group_id = resource.get('id')
+            if not group_id:
+
+                log.warning(
+                    f"Skipping group without ID: {resource.get('displayName', 'Unknown')}"
+                )
+
+                continue
+
+            # Get actual member count via Graph API
+            member_count = self.manager.get_group_member_count(group_id)
```

Comment:
For non-trivial group counts, running this inside a loop can lead to a large amount of API calls. It may be helpful to try something like [batching](https://learn.microsoft.com/en-us/graph/json-batching) those calls. It looks like that might help us chunk and cover 20 lookups per call instead of 1. Worth a shot.